pkg/apis/core/types.go

```diff
@@ -3504,6 +3506,21 @@ type ServiceSpec struct {
 	// cluster (e.g. IPv6 in IPv4 only cluster) is an error condition and will fail during clusterIP assignment.
 	// +optional
 	IPFamily *IPFamily
+
+	// topologyKeys is a preference-order list of topology keys which
+	// implementations of services should use to preferentially sort endpoints
+	// when accessing this Service. Topology keys must be valid label keys and
+	// at most 16 keys may be specified, and can't be used with externalTrafficPolicy=Local.
```

Comment:
`at most 16 keys may be specified.  topologyKeys can not be used at the same time as externalTrafficPolicy=Local.`